### PR TITLE
Fixed issue with ZCylinders extending outside geometry boundaries

### PR DESCRIPTION
**Pull request by samuelshaner**
This PR builds on PR #264 and fixes issue #269. The fix restructures the `Universe::getMin_()` and `Universe::getMin_BoundaryType()` methods by first checking to see if one of the cells in the universe contains a surface with a non-`BOUNDARY_NONE` boundaryType for the boundary in question (x-min, y-min, x-max, or y-max). If so, the corresponding min/max value or boundaryType for that geometry bounding surface is returned. If not, the min/max value is found by returning the min/max for all the cells contained in the universe; or, for the case of `Universe::getMin_BoundaryType()`, `BOUNDARY_NONE` is returned.

### CODE REVIEW

**Comment by geogunow** [comment]
@samuelshaner did this pass the tests on your local machine? I've run into issues in the past where Travis doesn't really work as expected. I'm curious if that's the case yet again or if there's just a bug in the code.

**Comment by samuelshaner** [comment]
Nope, I merged with wbinventor/multi-sim so most of the test failures are
inherited from that branch though. That PR should be merged before this
one.

On Wednesday, February 17, 2016, Geoffrey Gunow <notifications@github.com>
wrote:

> @samuelshaner <https://github.com/samuelshaner> did this pass the tests
> on your local machine? I've run into issues in the past where Travis
> doesn't really work as expected. I'm curious if that's the case yet again
> or if there's just a bug in the code.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mit-crpg/OpenMOC/pull/271#issuecomment-185050891>.
>


-- 
Regards,
Sam Shaner
Graduate Student
MIT Nuclear Science and Engineering
samuelshaner@gmail.com | (805)748-7684 | 24-607

File: tests/test_correct_fsr_volume/test_correct_fsr_volume.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        self.track_generator.setNumThreads(1)
+        self.track_generator.generateTracks()
+
+        # Correct FSR volumes with the appropriate number of threads
+        self.track_generator.setNumThreads(self.num_threads)
+        old_volume = self.track_generator.getFSRVolume(1)
+        self.track_generator.correctFSRVolume(1, 2.7)
+        new_volume = self.track_generator.getFSRVolume(1)
+
+        # Save results for one thread
```
Comment (geogunow) [inline tests/test_correct_fsr_volume/test_correct_fsr_volume.py]: Let's be consistent with one vs 1 here and in the comment a few lines up

**Comment by samuelshaner** [comment]
This was covered in a recently merged PR

**Pull request closed without merging**